packages/runtime/src/slate/types.ts

```diff
@@ -264,3 +269,21 @@ declare module 'slate' {
     Text: Text
   }
 }
+
+/**
+ * Custom types to extend prop support for React DOM canary elements.
+ */
+declare module 'react-dom' {
+  export interface PageLinkProps extends ComponentPropsWithoutRef<'link'> {
+    precedence?: 'reset' | 'low' | 'medium' | 'high'
+  }
+
+  export interface PageStyleProps extends ComponentPropsWithoutRef<'style'> {
+    precedence?: 'reset' | 'low' | 'medium' | 'high'
+    href?: string
+  }
+
+  export interface PageTitleProps extends Omit<ComponentPropsWithoutRef<'title'>, 'children'> {
+    children?: string
+  }
+}
```

Comment:
These shouldn't be in the types file for Slate, which is the library we use for rich text. Instead, we should place it in a file for React DOM or in the one file we need this types. Also, I don't think we should export prop types and instead should use augmentation to add support for the new props in the elements themselves.